web/src/app/monitors/monitor-create-edit/monitor-create-edit.component.ts

```diff
@@ -0,0 +1,108 @@
+import { Component, OnInit } from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { v4 as uuid } from 'uuid';
+
+// Third party modules
+import { MatSnackBar } from '@angular/material';
+import { ActivatedRoute, Router } from '@angular/router';
+import { Subscription } from 'rxjs';
+
+// Dashboard hub models and services
+import { ProjectService } from '../../core/services/index.service';
+import { MonitorModel, ProjectModel } from '../../shared/models/index.model';
+
+@Component({
+  selector: 'dashboard-monitor-create-edit',
+  templateUrl: './monitor-create-edit.component.html',
+  styleUrls: ['./monitor-create-edit.component.scss'],
+})
+export class MonitorCreateEditComponent implements OnInit {
+
+  private createEditSubscription: Subscription;
+  private projectSubscription: Subscription;
+
+  private id: string;
+  private monitorsList: MonitorModel[] = [];
+  private project: ProjectModel;
+  private uid: string;
+  public isEdit: Boolean = false;
+  public monitorForm: FormGroup;
+
+  constructor(
+    private form: FormBuilder,
+    private projectService: ProjectService,
+    private router: Router,
+    private route: ActivatedRoute,
+    private snackBar: MatSnackBar
+  ) { }
+
+  /**
+   * Lifecycle init method
+   */
+  ngOnInit(): void {
+    this.uid = this.route.snapshot.paramMap.get('uid');
+    this.monitorForm = this.form.group({
+      uid: uuid(),
+      name: [undefined, [Validators.required, Validators.minLength(3)]],
+      method: [undefined, Validators.required],
+      path: [undefined, Validators.required],
+      expectedCode: [undefined, Validators.required],
+      expectedText: [undefined],
+    });
+    this.projectSubscription = this.projectService
+      .findOneById(this.uid)
+      .subscribe((project: ProjectModel) => {
+        this.project = project;
+        this.monitorsList = this.project.monitors ? this.project.monitors : [];
+        this.id = this.route.snapshot.paramMap.get('id');
+        if (this.id) {
+          this.isEdit = true;
+          const filteredMonitor: MonitorModel[] = this.monitorsList.filter((monitor: MonitorModel) => {
```

Comment:
This component could be simplified if we move the monitor logic to a monitor service

Reply:
done